strict-csp-html-webpack-plugin/README.md

```diff
@@ -73,3 +73,68 @@ You can use additional options to configure the plugin:
 | `enableTrustedTypes` | `true`  | When `true`, enables [trusted types](https://web.dev/trusted-types) for additional protections against DOM XSS attacks. |
 
 
+
+## FAQ
+
+### Does this plugin protect my users from XSS attacks?
+
+It helps, but it's not enough.
+
+To protect your site from XSS, first of all, make sure to sanitize user input. Some frameworks do this by default.
+A strict CSP is an extra security layer (also called "defense-in-depth" technique) that helps mitigate XSS attacks in case there's a sanitization bug, or other XSS bug, in your application or dependency. A strict CSP prevents the execution of malicious scripts in case they make it to your application.
+
+### Where should I use this plugin?
+
+This plugin is best-suited for single-page applications. 
+If you have server-side logics, use a [nonce-based strict CSP](https://web.dev/strict-csp/#step-1:-decide-if-you-need-a-nonce-or-hash-based-csp) instead.
+
+### How does this plugin differ from [csp-html-webpack-plugin](https://www.npmjs.com/package/csp-html-webpack-plugin)?
+
+This plugin focuses on one thing: it sets up a [strict CSP](https://web.dev/strict-csp), that is, an efficient defense-in-depth mechanism against XSS attacks.
+
+csp-html-webpack-plugin is more flexible. If you're using a CSP for other purposes than XSS mitigation, check out [csp-html-webpack-plugin](https://www.npmjs.com/package/csp-html-webpack-plugin).
+
+### I already have a CSP on my site, with an allowlist*. Should I consider using this plugin?
+*An allowlist CSP looks as follows: `default-src https://cdn.example https://site1.example https://site2.example;`.
+
+It depends.
+
+If you're using your allowlist CSP purely to load scripts coming from a certain origin, you can keep using it.
+
+But if you're relying on your allowlist CSP for XSS protection: migrate to the more secure strict CSP approach, and consider using this plugin to help you do so.
+
+Allowlist-based CSP are not recommended anymore for XSS protection, because don't efficiently protect sites against XSS attacks: [research has shown that they can be bypassed](https://research.google/pubs/pub45542/). 🥲
+They're also harder to maintain!
+
+Instead, strict CSPs are now recommended, because they're both [more secure and easier to maintain than allowlist-based CSPs](https://web.dev/strict-csp/#why-a-strict-csp-is-recommended-over-allowlist-csps).
+
+This plugin automatically adds a strict CSP to your application.
+
+### Can this plugin slow down my site?
+
+See [issue #15](https://github.com/google/strict-csp/issues/15).
+
+### How does a strict CSP compare with subresource integrity (SRI)?
+
+SRI works well for static resources hosted on a CDN. 
+
+But it won't work anymore if scripts are changing, e.g. if they contain custom data, for example a twitter plugin or google analytics: in this case, you can't really use SRI because you don't know if/when they'll change their scripts.
```

Comment:
Maybe instead:
SRI can be used to ensure the integrity of scripts, e.g. to protect your site in case your CDN gets compromised.
However, SRI will not mitigate XSS caused by an injection vulnerability in your own site.

Reply:
Sounds good, used your suggestion.